### ISSUE TITLE
nscf bug for HSE calculation (same as the closed issue #2235)

### ISSUE THREAD

**Issue by iduygnay** [deepmodeling/abacus-develop#3323]
### Describe the bug

 NON-SELF CONSISTENT CALCULATIONS
terminate called after throwing an instance of 'cereal::Exception'
  what():  Failed to read 8 bytes from input stream! Read 0

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 0 PID 1949050 RUNNING AT xnode17
=   KILLED BY SIGNAL: 6 (Aborted)
===================================================================================

[issue.zip](https://github.com/deepmodeling/abacus-develop/files/13633424/issue.zip)

### Expected behavior

I know pyatb can solve this, and I'm trying to use pyatb. Just let you know there exists a bug.

### To Reproduce

The runscript is in Y_abacus343.sh, abacus is 3.4.3

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by PeizeLin** [comment]
Thank you for your feedback. We will try to fix this bug in the future. At present it is recommended to use pyatb for band calculation.

**Comment by zhubonan** [comment]
I am still experiencing similar issue when calculating HSE band structure using 3.10 LTS.

I found the following error in the STDERR after the NSCF calculation crashed:

```
terminate called after throwing an instance of 'cereal::Exception'
  what():  Failed to read 8 bytes from input stream! Read 0
terminate called after throwing an instance of 'cereal::Exception'
  what():  Failed to read 8 bytes from input stream! Read 0
terminate called after throwing an instance of 'cereal::Exception'
  what():  Failed to read 8 bytes from input stream! Read 0
terminate called after throwing an instance of 'cereal::Exception'
  what():  Failed to read 8 bytes from input stream! Read 0
terminate called after throwing an instance of 'cereal::Exception'
  what():  Failed to read 8 bytes from input stream! Read 0
terminate called after throwing an instance of 'cereal::Exception'
```

The SCF calculation appears to have successfully written the Hexx CSR files:

```
$ ls OUT.222/restart/
charge_0_0   charge_12_0  charge_2_0  charge_6_0  Eexx_0_0       HexxR11_0.csr  HexxR15_0.csr  HexxR5_0.csr  HexxR9_0.csr
charge_1_0   charge_13_0  charge_3_0  charge_7_0  HexxR0_0.csr   HexxR12_0.csr  HexxR2_0.csr   HexxR6_0.csr
charge_10_0  charge_14_0  charge_4_0  charge_8_0  HexxR10_0.csr  HexxR13_0.csr  HexxR3_0.csr   HexxR7_0.csr
charge_11_0  charge_15_0  charge_5_0  charge_9_0  HexxR1_0.csr   HexxR14_0.csr  HexxR4_0.csr   HexxR8_0.csr
```

Here is the SCF INPUT

```
pseudo_dir              .
orbital_dir             .

nspin                   1
calculation             scf
basis_type              lcao
ks_solver               genelpa
ecutwfc                 100 ###Energy cutoff needs to be tested to ensure your calculation is reliable.[1]
scf_thr                 1e-7
scf_nmax                100
gamma_only              0
symmetry                0
smearing_method         gauss
mixing_type             broyden
mixing_beta             0.4
exx_pca_threshold       1e-3
exx_c_threshold         1e-3
exx_dm_threshold         1e-3
#lspinorb                1
noncolin                0
smearing_sigma          0.001

dft_functional          hse
suffix                  222
out_band                 1
#init_chg                file
restart_save            1
#restart_load            1

```

NSCF INPUT

```
pseudo_dir              .
orbital_dir             .

nspin                   1
calculation             nscf
basis_type              lcao
ks_solver               genelpa
ecutwfc                 100 ###Energy cutoff needs to be tested to ensure your calculation is reliable.[1]
scf_thr                 1e-7
scf_nmax                100
gamma_only              0
symmetry                0
smearing_method         gauss
mixing_type             broyden
mixing_beta             0.4
exx_pca_threshold       1e-3
exx_c_threshold         1e-3
exx_dm_threshold         1e-3
#lspinorb                1
noncolin                0
smearing_sigma          0.001

dft_functional          hse
suffix                  222
out_band                 1
#init_chg                file
#restart_save            1
restart_load            1
```

**Comment by zhubonan** [comment]
Resolved - I used 3.10 which does not contain this PR.
PS: using `out_chg    1` instead of `restart_save`/`restart_load` works
